src/fklearn/causal/cate_learning/double_machine_learning.py

```diff
@@ -0,0 +1,159 @@
+from typing import List, Tuple
+
+import numpy as np
+import pandas as pd
+from sklearn import __version__ as sk_version
+from sklearn.base import RegressorMixin
+from sklearn.model_selection import KFold
+from sklearn.ensemble import GradientBoostingRegressor
+from toolz import curry
+
+from fklearn.common_docstrings import learner_pred_fn_docstring, learner_return_docstring
+from fklearn.training.utils import log_learner_time, expand_features_encoded
+from fklearn.types import LearnerReturnType
+
+
+def _cv_estimate(model: RegressorMixin,
+                 train_data: pd.DataFrame,
+                 features: List[str],
+                 y: str,
+                 n_splits: int) -> Tuple[pd.Series, List[RegressorMixin]]:
+
+    cv = KFold(n_splits=n_splits)
+    models = []
+    cv_pred = pd.Series(np.nan, index=train_data.index)
+
+    for train, test in cv.split(train_data):
+        m = model.fit(train_data[features].iloc[train], train_data[y].iloc[train])
+        cv_pred.iloc[test] = m.predict(train_data[features].iloc[test])
+        models += [m]
+
+    return cv_pred, models
+
+
+@curry
+@log_learner_time(learner_name='non_parametric_double_ml_learner')
+def non_parametric_double_ml_learner(df: pd.DataFrame,
+                                     feature_columns: List[str],
+                                     treatment_column: str,
+                                     outcome_column: str,
+                                     debias_model: RegressorMixin = GradientBoostingRegressor(),
+                                     debias_feature_columns: List[str] = None,
```

Comment:
Mutable default arguments must be avoided in python. Please use `None` instead, then 
```
if debias_model is None:
   debias_model = GradientBoostingRegressor()
```

Reply:
https://docs.python-guide.org/writing/gotchas/

Reply:
Nice to know!